sklearn/preprocessing/data.py

```diff
@@ -661,7 +667,11 @@ def partial_fit(self, X, y=None):
                                           self.n_samples_seen_)
 
         if self.with_std:
-            self.scale_ = _handle_zeros_in_scale(np.sqrt(self.var_))
+            if self.with_std is not True and self.with_std is not 2:
```

Comment:
How about `with_std not in (1, 2, True)`

Reply:
Sure, that's make sense.